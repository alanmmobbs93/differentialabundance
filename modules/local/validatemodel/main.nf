process VALIDATE_MODEL {
    tag "${models_yml}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/f5/f5ef116da0b04fc6359efa48f08b9acfc040a02d7a1ec62476c77b6db0fc1499/data' :
        'community.wave.seqera.io/library/r-jsonlite_r-optparse_r-tidyverse_r-yaml:18dc3fc2d990206d' }"

    input:
    tuple path(samplesheet)
    tuple val(meta), path(models_yml)

    output:
    tuple val(meta), path("pheno_table.csv"), emit: pheno_table
    tuple val(meta), path("designs.json")   , emit: designs
    tuple val(meta), path("warnings.json")  , emit: warnings, optional: true
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    validate_model.R \\
        --yml "${models_yml}" \\
        --samplesheet "${samplesheet}" \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(Rscript -e 'R.Version()\$version.string' | sed -n 's|\\[1\\] "R version \\(.*\\) (.*|\\1|p')
        tidyverse: \$(Rscript -e "cat(paste(packageVersion('tidyverse'), collapse='.'))")
        optparse: \$(Rscript -e "cat(paste(packageVersion('optparse'), collapse='.'))")
        yaml: \$(Rscript -e "cat(paste(packageVersion('yaml'), collapse='.'))")
        jsonlite: \$(Rscript -e "cat(paste(packageVersion('jsonlite'), collapse='.'))")
    END_VERSIONS
    """

    stub:
    """
    touch pheno_table.csv
    touch designs.json
    touch warnings.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(Rscript -e 'R.Version()\$version.string' | sed -n 's|\\[1\\] "R version \\(.*\\) (.*|\\1|p')
        tidyverse: \$(Rscript -e "cat(paste(packageVersion('tidyverse'), collapse='.'))")
        optparse: \$(Rscript -e "cat(paste(packageVersion('optparse'), collapse='.'))")
        yaml: \$(Rscript -e "cat(paste(packageVersion('yaml'), collapse='.'))")
        jsonlite: \$(Rscript -e "cat(paste(packageVersion('jsonlite'), collapse='.'))")
    END_VERSIONS
    """

}
