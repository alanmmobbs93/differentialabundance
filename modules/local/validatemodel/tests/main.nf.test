nextflow_process {

    name "Test Process VALIDATE_MODEL"
    script "../main.nf"
    process "VALIDATE_MODEL"
    config "./nextflow.config"

    tag "modules"
    tag "models"
    tag "yml"
    tag "samplesheet"

    test("Samplesheet - YML file - RNASEQ TEST") {

        when {
            params {
                observations_id_col = 'sample'
            }
            process {
                """
                // Samplesheet
                input[0] =
                    Channel.fromPath('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv', checkIfExists: true)

                // Models_yml sheet
                input[1] =
                    Channel.fromPath('${moduleDir}/tests/contrasts.yml', checkIfExists: true)
                        .map{ [ [id: "test_yml"], it ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
