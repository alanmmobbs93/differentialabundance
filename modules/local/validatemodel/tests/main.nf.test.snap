{
    "Samplesheet - YML file - RNASEQ TEST": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_yml"
                        },
                        "pheno_table.csv:md5,4969fce19ad1ea5b3cf124f4dcc6a403"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_yml"
                        },
                        "designs.json:md5,eb5b1a01da5d16b578f50d9bd7853ca0"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,bf359bcdd07f0bcc0a2c494ce2823950"
                ],
                "designs": [
                    [
                        {
                            "id": "test_yml"
                        },
                        "designs.json:md5,eb5b1a01da5d16b578f50d9bd7853ca0"
                    ]
                ],
                "pheno_table": [
                    [
                        {
                            "id": "test_yml"
                        },
                        "pheno_table.csv:md5,4969fce19ad1ea5b3cf124f4dcc6a403"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bf359bcdd07f0bcc0a2c494ce2823950"
                ],
                "warnings": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-19T16:59:52.146249557"
    }
}